js/ui/endlessButton.js

```diff
@@ -29,15 +30,30 @@ const EndlessButton = new Lang.Class({
     _setupTooltipText: function() {
 
         // Create a new tooltip label
-        this._label = new St.Label({ text: _("Show Desktop"),
-                                     style_class: 'app-icon-hover-label' });
+        this._label = new St.Label({ style_class: 'app-icon-hover-label' });
 
         this._labelOffsetX = 0;
         this._labelOffsetY = 0;
         this._label.connect('style-changed', Lang.bind(this, function(actor, forHeight, alloc) {
             this._labelOffsetX = this._label.get_theme_node().get_length('-label-offset-x');
             this._labelOffsetY = this._label.get_theme_node().get_length('-label-offset-y');
         }));
+
+        Main.overview.connect('page-changed', Lang.bind(this, this._updateHoverLabel));
+
+        this.actor.connect('destroy', Lang.bind(this, function() {
+            Main.overview.disconnect(this._updateHoverLabel);
```

Comment:
Pretty sure this doesn't do what you expect... `disconnect()` takes a signal ID, so the correct way to do this would be
```
let pageChangedId = Main.overview.connect(...);
this.actor.connect('destroy', function() {
    Main.overview.disconnect(pageChangedId);
});
```